ui/app/components/secret-engine/configure-aws.ts

```diff
@@ -5,78 +5,130 @@
+    const isValid = this.validate(leaseConfig);
+    if (!isValid) return;
+    // Check if any of the models' attributes have changed.
+    // If no changes to either model, transition and notify user.
+    // If changes to one model, save the model that changed and notify user.
+    // Otherwise save both models.
+    // Note: "backend" dirties model state so explicity ignore it here.
+    const leaseAttrChanged =
+      Object.keys(leaseConfig.changedAttributes()).filter((item) => item !== 'backend').length > 0;
+    const rootAttrChanged =
+      Object.keys(rootConfig.changedAttributes()).filter((item) => item !== 'backend').length > 0;
+
+    if (!leaseAttrChanged && !rootAttrChanged) {
+      this.flashMessages.info('No changes detected.');
+      this.transition();
+    }
+    if (rootAttrChanged) {
+      try {
+        yield rootConfig.save();
+        this.flashMessages.success(`Successfully saved ${id}'s root configuration.`);
+        this.transition();
+      } catch (error) {
+        this.errorMessageRoot = errorMessage(error);
+        this.invalidFormAlert = 'There was an error submitting this form.';
+      }
+    }
+    if (leaseAttrChanged) {
+      try {
+        yield leaseConfig.save();
+        this.flashMessages.success(`Successfully saved ${id}'s lease configuration.`);
+        this.transition();
+      } catch (error) {
+        // if lease config fails, but there was no error saving rootConfig: notify user of the lease failure with a flash message, save the root config, and transition.
+        if (!this.errorMessageRoot) {
+          this.flashMessages.danger(`Lease configuration was not saved: ${errorMessage(error)}`, {
+            sticky: true,
+          });
+          this.transition();
+        } else {
+          this.errorMessageLease = errorMessage(error);
+          this.flashMessages.danger(
+            `Configuration not saved: ${errorMessage(error)}. ${this.errorMessageRoot}`
+          );
+        }
+      }
+    }
   }
 
-  @action
-  saveLease(data: LeaseFields, event: Event) {
-    event.preventDefault();
-    this.args.saveAWSLease(data);
+  resetErrors() {
+    this.flashMessages.clearMessages();
+    this.errorMessageRoot = null;
+    this.invalidFormAlert = null;
+  }
+
+  transition() {
+    this.router.transitionTo('vault.cluster.secrets.backend.configuration', this.args.id);
+  }
+
+  validate(model: LeaseConfigModel) {
+    const { isValid, state, invalidFormMessage } = model.validate();
+    this.modelValidationsLease = isValid ? null : state;
+    this.invalidFormAlert = isValid ? '' : invalidFormMessage;
+    return isValid;
   }
 
   @action
-  handleTtlChange(name: string, ttlObj: TtlEvent) {
-    // lease values cannot be undefined, set to 0 to use default
-    const valueToSet = ttlObj.enabled ? ttlObj.goSafeTimeString : 0;
-    this.args.model.set(name, valueToSet);
+  onCancel() {
+    // clear errors because they're canceling out of the workflow.
```

Comment:
Do we also need to rollback the models?

Reply:
I had a model rollback and hit some issues because we're saving multiple models at once. Let me play with it again now that I'll separate out the save function.